app/src/main/java/com/paxus/pay/poslinkui/demo/entry/BaseEntryDialogFragment.java

```diff
@@ -160,30 +147,33 @@ protected void onEntryDeclined(long errCode, String errMessage){
         Toast.makeText(requireActivity(), errMessage, Toast.LENGTH_SHORT).show();
     }
 
-    @Subscribe(threadMode = ThreadMode.MAIN)
-    public void onGetEntryResponse(EntryResponseEvent event){
-        switch (event.action){
-            case EntryResponse.ACTION_ACCEPTED:
-                onEntryAccepted();
-                break;
-            case EntryResponse.ACTION_DECLINED:{
-                onEntryDeclined(event.code,event.message);
+    protected void onConfirmButtonClicked(){}
+
+    protected void executeEnterKeyEvent(){ onConfirmButtonClicked(); }
+    protected void executeBackPressEvent(){ sendAbort(); }
+
+    Observer<ResponseEvent> responseEventObserver = new Observer<ResponseEvent>() {
+        @Override
+        public void onChanged(ResponseEvent event) {
+            Logger.d(getClass().getSimpleName() + " receives " + event.action);
+            switch (event.action){
+                case EntryResponse.ACTION_ACCEPTED:
+                    onEntryAccepted();
+                    break;
+                case EntryResponse.ACTION_DECLINED:{
+                    onEntryDeclined(event.code,event.message);
+                }
             }
+            baseEntryDialogViewModel.resetResponseEvent();
```

Comment:
The observer is removed in next Fragment? What if next action is handled by DefaultUI?